### PR TITLE
Internationalization for flavours and skins

### PR DESCRIPTION
**Pull request by marrus-sh**
This first commit just allows for giving flavours and skins internationalized names via `names.yml`, and only touches the Ruby side of things. @beatrix-bitrot it's up to you whether you want javascript stuff to go in a separate PR or to be lumped into this one, but that's next.

### CODE REVIEW

**Comment by marrus-sh** [comment]
I went ahead and put the javascript internationalization stuff in there too; it adds a `locales` property to `themes.yml` which should point to the flavour's locale directory.

File: config/webpack/generateLocalePacks.js [hunk 1]
```diff
@@ -1,70 +1,66 @@
...
-import { setLocale } from 'locales';
-${glitchInject}
-setLocale({messages: mergedMessages, localeData: localeData});
-`;
-  fs.writeFileSync(localePath, localeContent, 'utf8');
-  outPaths.push(localePath);
-});
 
-module.exports = outPaths;
+import messages from '../../../${flavour.locales}/${locale.replace(/\.js$/, '')}';
```
Comment (marrus-sh) [inline config/webpack/generateLocalePacks.js]: If we make `${flavour.locales}` resolve relative to `app/javascript` if it doesn't start with a `/\.\.?\//` (aka follow babel resolution rules), then we should be able to simplify this import greatly.

**Comment by marrus-sh** [comment]
Don't merge this until I have a chance to bring it up to speed with the latest localization stuff (notably, the ja localization ;) )

**Comment by beatrix-bitrot** [comment]
oki np~

File: config/webpack/generateLocalePacks.js (hunk 1 above)
Comment (marrus-sh) [reply config/webpack/generateLocalePacks.js]: I kinda feel like this adds too much complexity for an import that nobody ever sees though :P

**Pull request merged**